--- a/java/ql/lib/semmle/code/java/security/LogInjection.qll
+++ b/java/ql/lib/semmle/code/java/security/LogInjection.qll
@@ -35,6 +35,13 @@ private class DefaultLogInjectionSink extends LogInjectionSink {
 private class DefaultLogInjectionSanitizer extends LogInjectionSanitizer instanceof SimpleTypeSanitizer
 { }
 
+/**
+ * A sanitizer for log injection that is defined via a barrier model
+ * in a data extension with the kind "log-injection".
+ */
+private class ExternalLogInjectionSanitizer extends LogInjectionSanitizer {
+  ExternalLogInjectionSanitizer() { barrierNode(this, "log-injection") }
+}
+
 private class LineBreaksLogInjectionSanitizer extends LogInjectionSanitizer {
   LineBreaksLogInjectionSanitizer() {
     logInjectionSanitizer(this.asExpr())
